src/Storages/MergeTree/MergeTreeSettings.cpp

```diff
@@ -225,6 +225,14 @@ void MergeTreeSettings::sanityCheck(size_t background_pool_tasks) const
             "The value of merge_selecting_sleep_slowdown_factor setting ({}) cannot be less than 1.0",
             merge_selecting_sleep_slowdown_factor);
     }
+
+    if (min_free_disk_ratio_to_throw_insert < 0.0 || min_free_disk_ratio_to_throw_insert > 1.0)
+    {
+        throw Exception(
+            ErrorCodes::BAD_ARGUMENTS,
+            "The value of setting min_free_disk_ratio_to_throw_insert ({}) must be between 0.0 and 1.0",
+            min_free_disk_ratio_to_throw_insert);
+    }
```

Comment:
Let's suppose we have some unusual value for this setting; we pass the sanity check, and now we have a default value for this setting, how do we catch this exception then?

Reply:
From what I have been testing, it seems that even if we pass the first sanity check (the one in `SettingsQuirks.cpp` we can still throw this exception: let's say we have default `min_free_disk_ratio_to_throw_insert = 0.0` and then we send to the server the query `create table b (id UInt64) engine = MergeTree order by id settings min_free_disk_ratio_to_perform_insert = 100`, then this query will fail with this exception and the table will not be created.

So basically my approach was to guarantee that there is always a default 'safe' value in the server config, and also prevent a individual table level to have an invalid value. Please let me know if this is reasonable or if I am just overcomplicating things, since I am quite new to this project.

Reply:
I think it will be better just to throw an exception without a sanity check because, as you said, if you get an exception if the data is incorrect, then a sanity check does not much sense🙂

Reply:
Ok I see, does make sense thanks!